src/execution/compiler/operator/hash_join_translator.cpp

```diff
@@ -216,7 +238,7 @@ void HashJoinTranslator::CheckJoinPredicate(WorkContext *ctx, FunctionBuilder *f
   auto *codegen = GetCodeGen();
 
   auto cond = ctx->DeriveValue(*join_plan.GetJoinPredicate(), this);
-  if (join_plan.RequiresLeftMark()) {
+  if (join_plan.GetLogicalJoinType() == planner::LogicalJoinType::LEFT_SEMI) {
```

Comment:
I thought this exists to prevent rows from the left table from appearing multiple times in the output. I didn't think this is appropriate for left outer joins, because the same left row can match with different right rows.

Reply:
You're right. I misread where this change was happening.